aplib.net-demo/Assets/Editor/GridPlacerEditor.cs

```diff
@@ -38,13 +40,12 @@ public override void OnInspectorGUI()
 
             gridPlacerSettings.ApplyModifiedProperties();
 
-            if (GUILayout.Button("Update Scene"))
-            {
-                GridPlacer gridPlacer = (GridPlacer)target;
-                foreach (Transform child in gridPlacer.transform) Destroy(child.gameObject);
+            if (!GUILayout.Button("Generate new level")) return;
```

Comment:
Not sure how this button works, but if you press it multiple times before pressing play, it generates multiple levels on top of each other. Maybe check and reset the grid before generating a level with this button